src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const correctedSeparator = separator;
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if ([null, undefined].includes(this[i])) {
+        this[i] = '';
+      }
```

Comment:
Don't repeat yourself: you write two same if statements, one here and one in the next cycle